diff -up hplip-3.14.6/prnt/hpcups/HPCupsFilter.cpp.logdir hplip-3.14.6/prnt/hpcups/HPCupsFilter.cpp
--- hplip-3.14.6/prnt/hpcups/HPCupsFilter.cpp.logdir	2014-06-05 13:33:05.426719552 +0200
+++ hplip-3.14.6/prnt/hpcups/HPCupsFilter.cpp	2014-06-05 13:33:05.432719467 +0200
@@ -604,7 +604,7 @@ int HPCupsFilter::processRasterData(cups
     char hpPreProcessedRasterFile[MAX_FILE_PATH_LEN]; //temp file needed to store raster data with swaped pages.
 
 
-    sprintf(hpPreProcessedRasterFile, "%s/hp_%s_cups_SwapedPagesXXXXXX",CUPS_TMP_DIR, m_JA.user_name);
+    snprintf(hpPreProcessedRasterFile, sizeof (hpPreProcessedRasterFile), "%s/hp_%s_cups_SwapedPagesXXXXXX",CUPS_TMP_DIR, m_JA.user_name);
 
 
     while (cupsRasterReadHeader2(cups_raster, &cups_header))
diff -up hplip-3.14.6/prnt/hpcups/SystemServices.cpp.logdir hplip-3.14.6/prnt/hpcups/SystemServices.cpp
--- hplip-3.14.6/prnt/hpcups/SystemServices.cpp.logdir	2014-06-05 13:33:05.432719467 +0200
+++ hplip-3.14.6/prnt/hpcups/SystemServices.cpp	2014-06-05 13:34:05.742861168 +0200
@@ -38,7 +38,7 @@ SystemServices::SystemServices(int iLogL
     if (iLogLevel & SAVE_OUT_FILE)
     {
         char    fname[MAX_FILE_PATH_LEN];
-        sprintf(fname, "%s/hpcups_%s_out_job%d_XXXXXX",CUPS_TMP_DIR, user_name, job_id);
+        snprintf(fname, sizeof(fname), "%s/hpcups_%s_out_job%d_XXXXXX",CUPS_TMP_DIR, user_name, job_id);
         
         createTempFile(fname, &m_fp);
         if (m_fp)
