diff -up hplip-3.12.4/installer/core_install.py.network-timeout hplip-3.12.4/installer/core_install.py
--- hplip-3.12.4/installer/core_install.py.network-timeout	2012-04-10 09:33:07.000000000 +0100
+++ hplip-3.12.4/installer/core_install.py	2012-08-23 12:27:02.815346648 +0100
@@ -1,4 +1,4 @@
-﻿# -*- coding: utf-8 -*-
+# -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2009 Hewlett-Packard Development Company, L.P.
 #
@@ -1586,7 +1586,7 @@ class CoreInstall(object):
             curl = utils.which("curl")
             if curl:
                 curl = os.path.join(curl, "curl")
-                cmd = "%s --output - --connect-timeout 5 --max-time 10 %s" % (curl, HTTP_GET_TARGET)
+                cmd = "%s --output - --connect-timeout 15 --max-time 10 %s" % (curl, HTTP_GET_TARGET)
                 log.debug(cmd)
                 status, output = self.run(cmd)
                 log.debug("curl returned: %d" % status)
@@ -1597,7 +1597,7 @@ class CoreInstall(object):
 
                 if ping:
                     ping = os.path.join(ping, "ping")
-                    cmd = "%s -c1 -W1 -w10 %s" % (ping, PING_TARGET)
+                    cmd = "%s -c1 -W3 -w30 %s" % (ping, PING_TARGET)
                     log.debug(cmd)
                     status, output = self.run(cmd)
                     log.debug("ping returned: %d" % status)
