diff -up hplip-3.14.6/prnt/hpcups/LJZjStream.cpp.uninit hplip-3.14.6/prnt/hpcups/LJZjStream.cpp
--- hplip-3.14.6/prnt/hpcups/LJZjStream.cpp.uninit	2015-01-21 12:10:23.836711750 +0000
+++ hplip-3.14.6/prnt/hpcups/LJZjStream.cpp	2015-01-21 12:12:24.026280851 +0000
@@ -681,14 +681,13 @@ DRIVER_ERROR LJZjStream::preProcessRaste
 	snprintf(hpEvenPagesFile, sizeof(hpEvenPagesFile), "%s/hp_%s_cups_EvenPagesXXXXXX",CUPS_TMP_DIR, m_pJA->user_name);
 	snprintf(hpOddPagesFile, sizeof(hpOddPagesFile), "%s/hp_%s_cups_OddPagesXXXXXX", CUPS_TMP_DIR, m_pJA->user_name);
 	
+        memcpy(&cups_header, firstpage_cups_header, sizeof(cups_page_header2_t));
 	if (1 != m_pJA->pre_process_raster || !firstpage_cups_header->Duplex){		                                  
 		return  NO_ERROR;                                  
     }    
 
     dbglog ("DEBUG: Getting Swaped Pages Raster.....\n");
 
-    memcpy(&cups_header, firstpage_cups_header, sizeof(cups_page_header2_t));
-
     //Create temp files to store odd, even and swaped pages.
 	fdEven = mkstemp (hpEvenPagesFile);
 	fdOdd = mkstemp (hpOddPagesFile);
