diff -up hplip-3.14.10/base/device.py.device_open hplip-3.14.10/base/device.py
--- hplip-3.14.10/base/device.py.device_open	2014-10-31 09:28:43.675181974 +0000
+++ hplip-3.14.10/base/device.py	2014-10-31 09:28:56.579250180 +0000
@@ -400,7 +400,7 @@ def makeURI(param, port=1):
                 mq = queryModelByURI(d)
 
                 result_code, device_id = \
-                    hpmudext.device_open(d, mq.get('io-mode', hpmudext.HPMUD_UNI_MODE))
+                    hpmudext.open_device(d, mq.get('io-mode', hpmudext.HPMUD_UNI_MODE))
 
                 if result_code == hpmudext.HPMUD_R_OK:
                     result_code, data = hpmudext.get_device_id(device_id)
